gce-scripts/scp_and_setup.sh

```diff
@@ -0,0 +1,139 @@
+#!/bin/bash
+
+# This file is broken up into three distinct parts.
+# The first part is uploading the packages to install
+# script to all machines, and kicking it off.
+# We do this first because 1) we need those packages to do anything else
+# and 2) it takes about 10 - 15 seconds per machine, so we parallelize it.
+
+# The second part is simply a check to see if the packages are finished
+# installing. We test the last machine in the list first because if that is
+# finished then all the other machines SHOULD also be finished. After we verify
+# that the last machine is finished, loop back through all of the machines and
+# make sure that they've all finished. If they haven't print out an error
+# warning for that machine and stop the whole process.
+# Takes the host file and the list of shards and
+# scps shards to hosts.
+# Also scps various scripts and installs pshtt
+# and all the necessary packages.
+# List of IPs, separated by line
+hosts_file='hosts.txt'
+# number of files that we need to cycle through
+num_files=$(ls -1q input_files/ | wc -l)
+# list of files; we do this deterministically
+# because then we can run this command across
+# other scripts and expect the same order of files.
+list_of_files=$(ls -1q input_files)
+# counter to keep track of which machine we're on (for logging purposes).
+i=1
+# We flip this bit if we find an error with any of the machines. This tells us
+# to stop the process so that the user can go by hand and fix the machine.
+error_with_packages=1
+
+# Upload script and install packages on all machines.
+# parallelized.
+################################################################
+for x in $list_of_files;
+do
+    # Grab the ip from hosts.txt that corresponds to the file number we are
+    # uploading.
+    # If we are uploading file #3 in the list, go to line 3 in the hosts file
+    # and upload to that ip.
+
+    machine=$(sed "${i}q;d" $hosts_file)
+    echo 'Now on '"${machine}"' number '$i
+    # Do not do strict host key checking so that you dont have to type "yes" for
+    # each machine.
+    echo 'Uploading packages_to_install.sh'
+    scp -i ~/.ssh/gce_pshtt_key -o "StrictHostKeyChecking no" packages_to_install.sh ubuntu@"${machine}":~/
```

Comment:
Is there a reason to do `-o "StrictHostKeyChecking no"` here, but not in the other commands? I see there is the comment above, but will it re-prompt at the next call?